ironic/drivers/modules/drac/raid.py

```diff
@@ -876,13 +968,84 @@ def _check_node_raid_jobs(self, task):
         task.upgrade_lock()
         self._delete_cached_config_job_id(node, finished_job_ids)
 
-        if not node.driver_internal_info['raid_config_job_ids']:
-            if not node.driver_internal_info.get('raid_config_job_failure',
-                                                 False):
-                self._resume_cleaning(task)
+        if not node.driver_internal_info.get('raid_config_job_failure',
+                                             False):
+            if 'raid_config_substep' in node.driver_internal_info:
+                if node.driver_internal_info['raid_config_substep'] == \
+                        'delete_foreign_config':
+                    controllers = list()
+                    jobs_required = False
+                    for controller_id in \
+                            node.driver_internal_info[
+                                'raid_config_parameters']:
+                        controller_cap = clear_foreign_config(
+                            node, controller_id)
+                        controller = {'raid_controller': controller_id,
+                                      'is_reboot_required':
+                                          controller_cap[
+                                              'is_reboot_required']}
+                        controllers.append(controller)
+                        jobs_required = jobs_required or controller_cap[
+                            'is_commit_required']
+
+                    if not jobs_required:
+                        node.driver_internal_info[
+                            'raid_config_substep'
+                        ] = 'change_disk_state'
+                        LOG.info(
+                            "No foreign drives detected,"
+                            "performing next substep : {}".format(
+                                node.driver_internal_info[
+                                    'raid_config_substep']))
```

Comment:
Instead of relying on the falling through to the next "if" statement and converting the drives, I would call into the factored out method directly here.  (See below comment)

Reply:
Nice idea. done